web/src/components/RegisteredForTaxesModal.tsx

```diff
@@ -97,7 +97,7 @@ export const RegisteredForTaxesModal = (props: Props): ReactElement => {
   };
 
   const showMunicipalityField = (): boolean => {
-    if (!userData) {
+    if (!userData || userData.profileData.nexusLocationInNewJersey === false) {
```

Comment:
I understand the temptation to combine these, since it already returns false there. But this is a guard clause, it's not intended to contain logic - just to check that userData actually exists before we perform the REAL logic on it. Due to typescript, it's necessary, but we know userData will pretty much never be undefined so you can think of just skipping past it. This logic should be added further down in the body of the function, rather than combined with the guard clause.